addons/resource.language.en_gb/resources/strings.po

```diff
@@ -22173,7 +22173,125 @@ msgctxt "#37057"
 msgid "Data chunk size used on SMB connections"
 msgstr ""
 
-#empty strings from id 37058 to 38010
+#empty strings from id 37058 to 37100
+
+#. Settings / Services "Caching" category label
+#: system/settings/settings.xml
+msgctxt "#37101"
+msgid "Caching"
+msgstr ""
+
+#. Description of category #37101 "Caching"
+#: system/settings/settings.xml
+msgctxt "#37102"
+msgid "This category contains settings that configure caching of local and network files, and internet streams."
+msgstr ""
+
+#. Setting #37103 "Buffer Mode"
+#: system/settings/settings.xml
+msgctxt "#37103"
+msgid "Buffer Mode"
+msgstr ""
+
+#. Description of setting #37103 "Buffer Mode"
+#: system/settings/settings.xml
+msgctxt "#37104"
+msgid "Configure the media content to buffer."
+msgstr ""
+
+#. Setting #37105 "Memory Size"
+#: system/settings/settings.xml
+msgctxt "#37105"
+msgid "Memory Size"
+msgstr ""
+
+#. Description of setting #37105 "Memory Size"
+#: system/settings/settings.xml
+msgctxt "#37106"
+msgid "The size of the memory buffer in Mbytes. Setting zero (0) forces buffering to disk, which is not recommend on flash storage devices (eMMC, SD cards, SSD)."
+msgstr ""
+
+#. Setting #37107 "Read Factor"
+#: system/settings/settings.xml
+msgctxt "#37107"
+msgid "Read Factor"
+msgstr ""
+
+#. Description of setting #37107 "Read Factor"
+#: system/settings/settings.xml
+msgctxt "#37108"
+msgid "This factor determines the max read rate in terms of Read Factor x avg. bitrate of a video file.[CR]Chosen value should be realistic for the bitrate of the content and the capacity of the connection e.g. if the max UHD Blu-Ray bitrate is 140 Mbit/s and the connection is 250 Mbit/s, the maximum Read Factor is 250/140 = 1.78x"
```

Comment:
I'm not sure of this one:

> Increasing the multiple forces Kodi to read more data into the cache with each read request

Amount of data read in each read request depends on Chunk Size

> Large multiples will delay playback start

This is not true from https://github.com/xbmc/xbmc/pull/23760 --> playback start depends only on Video Player queue time (> 8 seconds) In fact playback start when queue is full and cache starts filling (first byte received). See https://github.com/xbmc/xbmc/pull/23760#issuecomment-1722389237

Maybe:
"The read factor determines cache fill rate in terms of avg. bitrate of stream x Read Factor. Increasing this multiple, cache fills faster but more bandwidth is consumed. Use large multiples may cause more CPU usage on some devices, saturate the connection and worsen performance."

Reply:
The read-factor needs to be abandoned, of course not in the scope of this PR but the point is a thorough discussion should be done about the utility and application of this variable.

The code where rate is being manipulated using the read-factor is not too well understood by anybody and this is what I could gather from looking into history of the file `FileCache.cpp`. Perhaps @arnova has that particular knowledge about how that line of code is doing it's magic? If so, it would be better to make the line of code at least comprehensible in such a manner so that it's utility can be recognized in an obvious way.

In all of my testing removing that line of code had little impact on how soon the cache was being filled. Why? Because the memory cache is always limited and once the cache is full no further writes are possible which naturally rate-limits Kodi from reading more from the stream.

The other scenario where rate-limiting might be making sense to everyone that cares is this: let's suppose someone has set ~4GB or similar memory cache for Kodi -- if someone has done this indeed, we can freely assume they've the know-how to understand their internet will be utilized greedily to fill that kind of cache size, in fact. that's the whole point of having a big cache, the user wants Kodi to fill it greedily for their smooth playback needs. Had their internet were that good where a small cache would get filled instantly, they would've never made or at all needed to make the cache size bigger.

More than likely this concept of rate-limiting has been inspired from some internet video providers which provide their own app to the users, like Netflix. Now Netflix has the responsibility and drive to keep their servers available for all of their customers, so obviously they've rate-limiting baked in their app. However Kodi has no such responsibility of maintaining or caring for any such servers, it can indeed read all that the user wants it to read.

@fritsch I hope you'll read this. Would it make sense to make a PR for this?

Reply:
Sense for this PR: Get it in as fast as possible as is ... else bikeshedding will kill it :-)

And on topic: yes readrate is the next big thingy. It has different aspects depending on buffer is empty (filling it fast), but a flaw for filling a buffer back up, when readrate x factor does not fit in. This needs strong rework. I even think that this factor should not be exposed at all, but a better algorithm should make sure that the cache is always filled to a certain degree (PID regulator approach), so that we don't run in too much spikes.

Reply:
> Sense for this PR: Get it in as fast as possible as is ... else bikeshedding will kill it :-)

I'm waiting to someone approve it to merge 😉

Reply:
Yeah - I would do that right away, but I was not involved in the delta-discussions, which is why I would prefer if someone of those colleagues would give their ack.

Reply:
> The read-factor needs to be abandoned, of course not in the scope of this PR but the point is a thorough discussion should be done about the utility and application of this variable.
> 
> The code where rate is being manipulated using the read-factor is not too well understood by anybody and this is what I could gather from looking into history of the file `FileCache.cpp`. Perhaps @arnova has that particular knowledge about how that line of code is doing it's magic? If so, it would be better to make the line of code at least comprehensible in such a manner so that it's utility can be recognized in an obvious way.
> 
I'm not sure what "line of code" you're referring to?

Reply:
> The read-factor needs to be abandoned, of course not in the scope of this PR but the point is a thorough discussion should be done about the utility and application of this variable.

> The code where rate is being manipulated using the read-factor is not too well understood by anybody and this is what I could gather from looking into history of the file

> In all of my testing removing that line of code had little impact on how soon the cache was being filled

Seems you suggest all code is not explained/understood (or you not understood) should be removed :)

This is only an example of how works Read Factor: as mentioned determines cache fill rate... 

It's just a slightly extreme example with very large cache sizes and "low" read factors.

**Read Factor = 4.0x** (took 34s to fill 191 MB of cache)

https://github.com/xbmc/xbmc/assets/58434170/feed767e-672f-48b2-b439-22d29b93b30e


**Read Factor = 1.5x** (took 2m 39s to fill 191 MB of cache) recording split in two parts in two parts

https://github.com/xbmc/xbmc/assets/58434170/c4c5fbd4-f116-487a-abfc-7d558c364dc9

https://github.com/xbmc/xbmc/assets/58434170/291d4864-d780-477d-b0a1-f2113437e748

Reply:
Kodi is a player, not a downloader. The mission of a downloader may be to download at the highest possible speed. A player must "download" at just the right speed to play smoothly. We should prioritize the playback thread more than the download thread and of course should not waste/use more CPU than necessary to download. In some devices, while playing 4K Dolby Vision, the CPU can be at 60 -70% if an Ethernet connection is also managed, downloading at 800 Mbps, CPU peaks can be generated that end up causing frame drops/skips. Limiting the download speed to only 150 - 200 Mbps (for a 100 Mbps stream) avoids unnecessary CPU spikes and is not required cache 100% filled to start playback.

Reply:
I have very little free time and this is all off topic because this PR is not intended to change the functionality. I'm not going to waste any more time on this discussion.

Reply:
Would've been better if you had not jumped in at all with your needless diatribe about how, "Kodi isn't a downloader" shtick. I was here looking for an explanation/reasoning, not from you but from the devs that maybe-probably understand that code section better than anybody else.

But seems like you've wasted not only "the very little free time" of yours, but very valuable time of mine as well, and almost everyone else's time who will end up reading this garbage because you couldn't control yourself from polluting the thread with conjectures.

I've seen this behavior from you in the past and I highly recommend that you start developing the habit of comprehensively reading & understanding a post's content thoroughly before hitting that reply button.